keyring.go

```diff
@@ -163,12 +169,14 @@ func (k KeyRing) VerifyJSONs(ctx context.Context, requests []VerifyJSONRequest)
 			// This means that we've checked every JSON object we can check.
 			return results, nil
 		}
+		logger.Infof("Requesting %d keys from %s", len(keyRequests), fetcher.FetcherName())
 		// TODO: Coalesce in-flight requests for the same keys.
 		// Otherwise we risk spamming the servers we query the keys from.
-		keysFetched, err := k.KeyFetchers[i].FetchKeys(ctx, keyRequests)
+		keysFetched, err := fetcher.FetchKeys(ctx, keyRequests)
 		if err != nil {
 			return nil, err
 		}
+		logger.Infof("Got %d keys from fetcher: verifying", len(keysFromDatabase))
```

Comment:
I think maybe we should try and use `WithFields` for things like this, to make it easier to navigate logs (if we start using non-file based logging)

Also in that vein: it might be good to include the `FetcherName` in all log entries? I think you can just do the following to get it included in all log entries:

```golang
logger := util.GetLogger(ctx).WithFields(log.Fields{
    "fetcher_name": fetcher.FetcherName(),
})
```

Reply:
OTOH, I guess `FetcherName` isn't the most useful of bits of information to search from, but I think maybe we should just get into the habit? Thoughts?

Reply:
bleh. I think that once I get into fetchers that delegate to other fetchers (as my forthcoming perspectives thing will), it's going to be more confusing than helpful.

Reply:
Is that about adding `FletcherName` to this particular log line, or when including it using `WithFields`

I'm tempted to have a policy of never formatting log lines and instead use `WithFields` for any dynamic content. (This also has the nice side effect of being able to easily grep for a specfic log line)

Reply:
oh I see. I thought you meant creating a whole new logger for the context. Yes, I'll have a go at that.